From 17c705fb3fefac4ce1718e90deb8623768bec4b0 Mon Sep 17 00:00:00 2001
From: Hariprasad Kelam <hkelam@marvell.com>
Date: Fri, 11 Jun 2021 15:14:00 +0530
Subject: [PATCH 1583/1921] octeontx2-af: Read mac address from correct offset

DMAC CAM is per CGX/RPM block dont pass lmac_id while
accessing the same.

Fixes: d3b2a96c0f9e ("octeontx2-af: add multicast filter support")

Change-Id: Ibc262b6200f58a94c8eda05ebfa1778a76b5ecfb
Signed-off-by: Hariprasad Kelam <hkelam@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/53997
Tested-by: Sunil Kovvuri Goutham <sgoutham@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <sgoutham@marvell.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/af/cgx.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/cgx.c b/drivers/net/ethernet/marvell/octeontx2/af/cgx.c
index 8f389a518e04..551f2085b2ac 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/cgx.c
+++ b/drivers/net/ethernet/marvell/octeontx2/af/cgx.c
@@ -416,8 +416,7 @@ int cgx_lmac_addr_del(u8 cgx_id, u8 lmac_id, u8 index)
 	index = id * lmac->mac_to_index_bmap.max + index;
 
 	/* Read MAC address to check whether it is ucast or mcast */
-	cfg = cgx_read(cgx_dev, lmac_id,
-		       (CGXX_CMRX_RX_DMAC_CAM0 + (index * 0x8)));
+	cfg = cgx_read(cgx_dev, 0, (CGXX_CMRX_RX_DMAC_CAM0 + (index * 0x8)));
 
 	cfg2mac(cfg, mac);
 	if (is_multicast_ether_addr(mac))
-- 
2.31.1

